lib/SILOptimizer/Transforms/ArrayElementValuePropagation.cpp

```diff
@@ -154,31 +181,67 @@ bool ArrayAllocation::mapInitializationStores() {
 /// allocated.
 bool ArrayAllocation::isInitializationWithKnownElements() {
   ArraySemanticsCall Uninitialized(Alloc, "array.uninitialized");
-  if (Uninitialized &&
-      (ArrayValue = Uninitialized.getArrayValue()) &&
-      (ElementBuffer = Uninitialized.getArrayElementStoragePointer()))
+  if (Uninitialized && (ArrayValue = Uninitialized.getArrayValue()) &&
```

Comment:
The reason why is that you want to make clear to the reader that what you are really trying to do is:

```
if (!Condition) {
  EARLY_EXIT
}

DO_WORK
```

And not:

```
if (CONDITION) {
  DO_WORK_1
  return
}
DO_WORK_2
```